sql-plugin/src/main/scala-2.13/com/nvidia/spark/rapids/Arm.scala

```diff
@@ -0,0 +1,232 @@
+/*
+ * Copyright (c) 2020-2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.nvidia.spark.rapids
+
+import scala.collection
+import scala.collection.{immutable, mutable}
+import scala.collection.mutable.{ArrayBuffer, ListBuffer}
+import scala.util.control.ControlThrowable
+
+import com.nvidia.spark.rapids.RapidsPluginImplicits._
+
+/** Implementation of the automatic-resource-management pattern */
+object Arm {
```

Comment:
Can we do some refactoring to minimize the differences between the various 2.13 and 2.12 files? I don't see having two sets of some of these really big files is going to be maintainable in the long term.

Reply:
Refactored most of these